sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -761,14 +762,36 @@ public static long ConvertTimeSpanToMilliseconds(TimeSpan timeSpan)
         /// <returns>String version of the data</returns>
         public static string ToHex(byte[] data, bool lowercase)
         {
-            StringBuilder sb = new StringBuilder();
-
-            for (int i = 0; i < data.Length; i++)
+#if NET8_0_OR_GREATER
+            if (!lowercase)
             {
-                sb.Append(data[i].ToString(lowercase ? "x2" : "X2", CultureInfo.InvariantCulture));
+                return Convert.ToHexString(data);
             }
+#endif
+            char[] chars = ArrayPool<char>.Shared.Rent(data.Length * 2);
 
-            return sb.ToString();
+            try
+            {
+                Func<int, char> converter = lowercase ? (Func<int, char>)ToLowerHex : (Func<int, char>)ToUpperHex;
+
+                for (int i = 0; i < data.Length; i++)
```

Comment:
Out of curiosity have you tried reversing the loop to elide bound checks?

Also could this be a candidate for `string.Create` since that would take care of the pooling already?

Reply:
Thanks for the suggestions @danielmarbach ! Both were very helpful.